src/plasmapy/simulation/particle_tracker/particle_tracker.py

```diff
@@ -584,6 +596,147 @@ def inner_Bethe_stopping(v, n_e):
         self._stopping_method = method
         self._stopping_power_interpolators = stopping_power_interpolators
 
+    # TODO: create a method to validate that these quantities have shape ngrids
+    # TODO: is this signature too much?
+    # TODO: find a way to create a type for the signature of the Callable parameters
+    #  (preferably using only one expression)
+    @particle_input
+    @validate_quantities
+    def add_scattering(
+        self,
+        target: ParticleLike,
+        differential_cross_section: (
```

Comment:
I don't entirely understand where these callables come from, but here's an idea. 
I suggest: 

1) Make all of these parameters optional keywords to this function along with a 'method' keyword that chooses between the two. Make clear in the docstring which keywords are for which method, and which are for both. 

2) For the given method, ensure that the associated keywords are set, match the grid size, etc. 

3) Create the callable inside this function?

Reply:
Just finished this, let me know what you think of how everything is documented :)

Reply:
Two concerns still: 

1) The callables don't know what scattering type they are (?), so there's the potential for conflict. I still think it would be better, for the limited number of modes we have, to just have the callable selected internally based on a keyword that selects which mode to use? 

2) You should add all the other keywords as parameters, e.g. `T_s`, `n_s` to explain in detail what each one is. The docstring for the parameter can say `Only used for routines ....` for each parameter so it's clear which are required when. But we need to clearly define what they all are, and that's easier when they are all  listed as separate parameters in the normal format.